sdk/azidentity/device_code_credential.go

```diff
@@ -32,19 +32,30 @@ type DeviceCodeCredentialOptions struct {
 // For more information on the device code authentication flow see: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-device-code.
 type DeviceCodeCredential struct {
 	client       *aadIdentityClient
-	tenantID     string       // Gets the Azure Active Directory tenant (directory) ID of the service principal
-	clientID     string       // Gets the client (application) ID of the service principal
-	userPrompt   func(string) // Sends the user a message with a verification URL and device code to sign in to the login server
-	refreshToken string       // Gets the refresh token sent from the service and will be used to retreive new access tokens after the initial request for a token. Thread safety for updates is handled in the AuthenticationPolicy since only one goroutine will be updating at a time
+	tenantID     string                  // Gets the Azure Active Directory tenant (directory) ID of the service principal
+	clientID     string                  // Gets the client (application) ID of the service principal
+	userPrompt   func(DeviceCodeMessage) // Sends the user a message with a verification URL and device code to sign in to the login server
+	refreshToken string                  // Gets the refresh token sent from the service and will be used to retreive new access tokens after the initial request for a token. Thread safety for updates is handled in the AuthenticationPolicy since only one goroutine will be updating at a time
+}
+
+// DeviceCodeMessage is used to store device code related information to help the user login and allow the device code flow to continue
+// to request a token to authenticate a user.
+type DeviceCodeMessage struct {
+	// User code returned by the service.
+	UserCode string `json:"user_code"`
+	// Verification URL where the user must navigate to authenticate using the device code and credentials.
+	VerificationURL string `json:"verification_uri"`
+	// User friendly text response that can be used for display purposes.
+	Message string `json:"message"`
 }
 
 // DefaultDeviceCodeCredentialOptions provides the default settings for DeviceCodeCredential.
 func DefaultDeviceCodeCredentialOptions() DeviceCodeCredentialOptions {
 	return DeviceCodeCredentialOptions{
 		TenantID: organizationsTenantID,
 		ClientID: developerSignOnClientID,
-		UserPrompt: func(s string) {
-			fmt.Println(s)
+		UserPrompt: func(dc DeviceCodeMessage) {
+			fmt.Println(dc.Message)
```

Comment:
Should update the doc comment for `DefaultDeviceCodeCredentialOptions()` indicating that it defaults to `stdout` (in fact, should probably document all the default values).